ui/components-react/src/components-react/newEditors/interop/EditorInterop.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import type {
+  Primitives,
+  PrimitiveValue,
+  PropertyRecord,
+} from "@itwin/appui-abstract";
+import { PropertyValueFormat } from "@itwin/appui-abstract";
+import type {
+  BooleanValue,
+  DateValue,
+  EnumValue,
+  InstanceKeyValue,
+  NumericValue,
+  TextValue,
+} from "../values/Values.js";
+import { Value as NewEditorValue } from "../values/Values.js";
+import type { OldEditorMetadata } from "./Metadata.js";
+
+/**
+ * Interop utilities for converting between old and new editor values.
+ * @internal
+ */
+export namespace EditorInterop {
+  /**
+   * @internal
+   */
+  export function getMetadataAndValue(propertyRecord: PropertyRecord): {
+    metadata: OldEditorMetadata | undefined;
+    value: NewEditorValue | undefined;
+  } {
+    const baseMetadata: Omit<OldEditorMetadata, "type"> = {
+      preferredEditor: propertyRecord.property.editor?.name,
+      params: propertyRecord.property.editor?.params,
+      extendedData: propertyRecord.extendedData,
+      enum: propertyRecord.property.enum,
+      typename: propertyRecord.property.typename,
+    };
+
+    const primitiveValue = propertyRecord.value as PrimitiveValue;
+    switch (propertyRecord.property.typename) {
+      case "text":
+      case "string":
+        return {
+          metadata: {
+            ...baseMetadata,
+            type: "string",
+          },
+          value: {
+            value: (primitiveValue.value as string) ?? "",
+          } satisfies TextValue,
+        };
+      case "dateTime":
+      case "shortdate":
+        return {
+          metadata: {
+            ...baseMetadata,
+            type: "date",
```

Comment:
I believe the difference between `dateTime` and `shortDate` in the old system was that the former also accepted time, while the latter only accepted date. Do we have that distinction in the new system?